### ISSUE TITLE
Crash on iPhone device (iOS 10.0.1), when options append `ResizingImageProcessor`

### ISSUE THREAD

**Issue by Limon-O-O** [onevcat/Kingfisher#459]
When I want to resize the `https://bubble-m.b0.upaiyun.com/thumbnail/3/79a0c0f859cc46fd9463dc7377c1318c` image on my iPhone device with `ResizingImageProcessor `, 
and crash at `context.draw(self, in: CGRect(x: 0, y: 0, width: width, height: height))`, 
but not crash on simulator.

```swift
let url = URL(string: "https://bubble-m.b0.upaiyun.com/thumbnail/3/79a0c0f859cc46fd9463dc7377c1318c")!

var options: KingfisherOptionsInfo = [.transition(.fade(0.3)), .backgroundDecode, .scaleFactor(UIScreen.main.scale)]

let processor = ResizingImageProcessor(targetSize: CGSize(width: 42.0, height: 42.0))
        options.append(.processor(processor))

_ = (cell as! CollectionViewCell).cellImageView.kf.setImage(with: url,
                                                   placeholder: nil,
                                                   options: options,
                                                   progressBlock: nil,
                                                   completionHandler: nil)
```

**Comment by onevcat** [comment]
Thanks for reporting! I'll take a look at it soon.

**Comment by onevcat** [comment]
Should be fixed in version 3.1.1. Let me know it does not work for you!
